lib/Array.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * A helper type to ensure an array has at least 1 element.
+ */
+export type NonEmptyArray<T> = [T, ...T[]]
+
+export namespace ArrayUtils {
+  /**
+   * A typesafe way for removing all null/undefined values from an array.
+   */
+  export const removeOptionals = <T>(arr: (T | null | undefined)[]) => {
+    return arr.reduce((acc, curr) => (curr ? [...acc, curr] : acc), [] as T[])
```

Comment:
We could also use .filter() to remove the null/undefined values

Reply:
`filter` doesn't actually change the type however, the return would still be `(T | undefined | null)[]`, which is why a `reduce` was needed.